text/MichałZwonek/opis.tex

```diff
@@ -0,0 +1,173 @@
+  \item Dopóki $S$ jest niepusty, wybierz $s,t$ z $S$ maksymalizujące $ov(s,t)$.\\
+  Dla $s\neq t$, niech $S := (S \setminus \{s, t\}) \cup \{\langle s, t \rangle\}, T := T$.\\
+  Dla $s = t$, niech $S := S \setminus \{s\}$, a $T := T \cup \{s\}$. 
+  \item Jako wspólne nadsłowo zwróc konkatenację słów ze zbioru $T$.
+\end{enumerate}
+
+Rozważmy teraz graf $G'_S$, analogiczny jak $G_S$, ale dla każdej krawędzi mamy jako wagę $ov(i,j)$ zamiast $d(i,j)$.
+Zauważmy, że dla dowonlego pokrycia cyklowego grafu $G_S$ oraz tego samego pokrycia cyklowego grafu $G'_S$ mamy, że 
+suma wag z pokrycia cyklowego z tych dwóch grafów równa się sumie długości słów w $S$. Jest tak ponieważ $\forall s,t: ov(s,t) + d(s,t) =  |s|$. Czyli maksymalne pokrycie cyklowe grafu $G'_S$, będzie tożsame z
+minimalnym pokryciem cyklowym grafu $G_S$. Okazuje się, że \textbf{MGREEDY} łącząc dwa słowa $s\neq t \in S$ w jedno, de facto 
+wybiera krawędź w grafie $G'_S$ między $s_\alpha := last(s)$, a $s_\beta := first(t)$ takie, że $ov(\alpha, \beta)$ jest maksymalne wśród wszystkich wolnych $ov(i,j)$.
+
+Wystarczy teraz tylko pokazać, że taka zachłanna strategia dla tego rodzaju grafu konstruuje pokrycie cyklowe o maksymalnej wadze, co będzie oznaczało, że \textbf{MGREEDY} jest niegorszy niż \textbf{CONCATCYCLES}, czyli aproksymuje ze współczynnikiem 4. To też pokazali w swojej oryginalnej pracy Blum et al.
+
+\section*{Algorytm \textit{TGREEDY}}
+
+Algorytm \textbf{TGREEDY} najpierw wykonuje kroki od 1 do 2 z algorytmu \textbf{MGREEDY}, a następnie na uzyskanym zbiorze $T$, który odpowiada
+maksymalnemu pokryciu cyklowemu grafu $G'_S$, wykonuje algorytm \textbf{GREEDY}. Aby pokazać, że \textbf{TGREEDY} ma pożądany współczynnik aproksymacj potrzebujemy wprowadzić parę dodatkowych definicji i lematów.
+
+\begin{definition} Niech $\equiv$ będzie relacją równoważności dla słów, taką, że $s \equiv t \Leftrightarrow \exists u,v: s = uv, t = vu$. 
+Wtedy, jeżeli $c$ to cykl na wierzchołkach (słowach) $i_0, ..., i_r$ to 
+$strings(c)$ jest klasą równoważności $[pref(i_0, i_1)pref(i_1,i_2)...pref(i_{r-1},i_0)]_\equiv$.\\
+Ponadto, $strings(c, i_k)$ zdefiniujemy jako
+$pref(i_k, i_{k+1})...pref(i_{k-1},i_{k})$, gdzie indeksowanie jest modulo $r$. Oczywiście, $strings(c, i_k) \in strings(c)$.
+\end{definition}
+\begin{definition}
+Niech $w(c)$, będzie wagą cyklu, czyli $w(c) = |s|, s\in strings(c)$. Dla wygody będziemy pisać, że $s\in c$ zamiast $s\in strings(c)$.
+\end{definition}
+
+\begin{lemma}\label{lemma9}
+  Niech $c_1,c_2$ będą dwoma cyklami w minimalnym pokryciu cyklowym $C$, z $s_1\in c_1$ oraz $s_2 \in c_2$ Wtedy, $ov(s_1,s_2) \leq w(c_1) + w(c_2)$. 
+\end{lemma}
+\begin{lemma}\label{claim4}
+Nadsłowo $\langle s_{i_0},...,s_{i_{r-1}} \rangle $ jest podsłowem $strings(c, i_0)s_{i_0}$.
+\end{lemma}
+
+\begin{theorem}
+\textbf{TGREEDY} dla zbioru $S = \{s_1, ..., s_m \}$ zwraca nadsłowo $s$ nad $S$ o długości co najwyżej $3\cdot OPT(S)$.
+\end{theorem}
+\begin{proof}
```

Comment:
For your information: I think that a bit simpler proof you may found in Vazirani (Approximation Algorithms) or in https://www.mimuw.edu.pl/~mucha/teaching/aa2008/ss.pdf - but this one is altogether sufficient.

Reply:
Thank you for the information, I'll take a look at the proof you provided. Thanks!